themes-default/slim/views/vue-components/selectListUi.mako

```diff
@@ -0,0 +1,64 @@
+<script type="text/x-template" id="select-list">
+    <div class="select-list">
+            <link rel="stylesheet" type="text/css" href="css/vue/selectlistui.css?${sbPID}" />
+            <ul>
+            <li v-for="item of editItems">
+                <input style="display: inline-block" type="text" @change="sendValues" v-model="item.value"/>
+                <img style="display: inline-block" src="images/no16.png" alt="N" width="16" height="16" @click="deleteItem(item)">
+            </li>
+            <div class="new-item">
+                <input style="display: inline-block" type="text" v-model="newItem"/>
+                <img style="display: inline-block" src="images/pencil_add.png" alt="N" width="16" height="16" @click="addItem">
+            </div>
+    </div>
+</script>
+<script>
+    // register the component
+    Vue.component('select-list', {
+        template: '#select-list',
+        props: ['listItems'],
+        mounted() {
+            this.editItems = this.sanitize(this.listItems);
+        },
+        data: function() {
+            return {
+                editItems: [],
+                newItem: '',
+                indexCounter: 0
+            }
+        },
+        methods: {
+            sendValues: function() {
+                this.$emit('change', this.editItems);
+            },
+            addItem: function(evt) {
+                this.editItems.push({id: this.indexCounter, value: this.newItem});
+                this.indexCounter += 1;
+                this.newItem = '';
+                this.sendValues();
+            },
+            deleteItem: function(item) {
+                this.editItems = this.editItems.filter(e => e !== item);
+                this.newItem = item.value;
+                this.sendValues();
+            },
+            sanitize: function(items) {
+                if (!items) {
+                    return [];
+                }
+                if (items.filter(item => typeof(item) === 'string').length) {
+                    return this.transformToIndexedObject(items);
+                } else {
+                    return items;
+                }
+            },
+            transformToIndexedObject: arrayOfStrings => {
```

Comment:
I'd suggest switching this to a map function.

Reply:
didn't know you could also inject the index into the map callback. Will do!

Reply:
done